internal/extension/types/hooks.go

```diff
@@ -0,0 +1,35 @@
+package types
+
+import (
+	"github.com/envoyproxy/gateway/internal/ir"
+
+	"github.com/envoyproxy/gateway/internal/extension/proto"
+	xdsTypes "github.com/envoyproxy/gateway/internal/xds/types"
+)
+
+type HookType string
+
+const (
+	XDSHook          HookType = "xds"
+	InfraManagerHook HookType = "infra-manager"
+)
+
+// XDSHook specifies the entrypoints for extensions to hook into the XDS
+// translation pipeline.
+type XDSHookClient interface {
+	// PostHTTPListenerTranslation is the entrypoint for modifying XDS after processing
+	// the HTTP listener.
+	PostHTTPListenerTranslation(eCtx *proto.HTTPListenerExtensionContext, tCtx *xdsTypes.ResourceVersionTable) error
+}
+
+// InfraManagerHook species for the entrypoints for extensions to hook into
+// the InfraManager translation pipeline.
+//
+// WARNING: This interface isn't finalized at all and subject to change
+type InfraManagerHookClient interface {
```

Comment:
Let's drop Infra stuff at this time. Yes, we can explore it once we have a v0 but I would rather it not distract from getting a working v0